src/storage/core.rs

```diff
@@ -997,12 +992,23 @@ where
     /// Delete entries with matching key
     /// # Errors
     /// Fails after any disk IO errors.
-    pub async fn delete(&self, key: impl AsRef<K>, only_if_presented: bool) -> Result<u64> {
+    pub async fn delete(&self, key: impl AsRef<K>, timestamp: BlobRecordTimestamp, only_if_presented: bool) -> Result<u64> {
+        self.delete_with_optional_meta(key, timestamp, None, only_if_presented).await
+    }
+
+    /// Delete entries with matching key. Appends metadata to deletion record
+    /// # Errors
+    /// Fails after any disk IO errors.
+    pub async fn delete_with(&self, key: impl AsRef<K>, timestamp: BlobRecordTimestamp, meta: Meta, only_if_presented: bool) -> Result<u64> {
```

Comment:
Isn't this method redundant, it just accepts subset of `delete_with_optional_meta` arguments

Reply:
I keep it consistent with the `write` functions. We also have 3 of them: public `write`, public `write_with` and internal `write_with_optional_meta`